orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -53,19 +53,37 @@ import static java.time.ZoneOffset.UTC
 class BakeStage implements StageDefinitionBuilder {
 
   public static final String PIPELINE_CONFIG_TYPE = "bake"
+  public static final String BAKE_PAUSE_TOGGLE = "stages.bake-stage.pause"
 
-  @Autowired
-  RegionCollector regionCollector
+  private RegionCollector regionCollector
+  private Clock clock
+  private DynamicConfigService dynamicConfigService
 
   @Autowired
-  Clock clock = systemUTC()
+  BakeStage(RegionCollector regionCollector, DynamicConfigService dynamicConfigService, Clock clock = systemUTC()) {
+    this.regionCollector = regionCollector
+    this.clock = clock
+    this.dynamicConfigService = dynamicConfigService
+  }
 
   @Override
   void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder builder) {
     if (isTopLevelStage(stage)) {
       builder
         .withTask("completeParallel", CompleteParallelBakeTask)
     } else {
+      if (dynamicConfigService.isEnabled(BAKE_PAUSE_TOGGLE, false)) {
+        log.info("Baking is currently paused. Adding pause task to ${stage.name} stage.")
+        stage.context.put("pauseToggle", BAKE_PAUSE_TOGGLE)
+        builder.withTask("pauseBake", ToggleablePauseTask)
+
+        if (stage.context.containsKey("stageTimeoutMs")) {
```

Comment:
instead of this i would put this warning in `ToggleablePauseTask`, just override the `onTimeout` method.

That way, if we will only see these messages if the bake actually timeout out - this will make things easier to find in logs if we need to do a surgical rescue mission.
There is, of course, the case of the stage being unpaused with, say, 1 minute left on the clock so the bake task itself will fail, which is unfortunate, but again, we can search for that in the logs pretty easily.

I think, ideally, `RunTaskHandler` doesn't account for this pause time in timeout calculations, but that's probably not worth the effort
Ideall

Reply:
Wait, I'm confused... I thought `Task.onTimeout()` would be called if the task's own `getTimeout()` had elapsed, not if the parent stage timed out. Which one is it?

Reply:
The currently executing task will have `onTimeout` [called](https://github.com/spinnaker/orca/blob/master/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt#L115) if the stage times out (due to `stageTimeoutMs`)

Reply:
I see. It checks for _both_ stage and task timeout. Got it.